pkg/block/mdfetcher.go

```diff
@@ -0,0 +1,103 @@
+package block
+
+import (
+	"context"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/ulid"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+	"github.com/thanos-io/thanos/pkg/objstore"
+)
+
+type MetadataFetcher struct {
+	cachedBlockMDs map[ulid.ULID]*metadata.Meta
+	rwLock         *sync.RWMutex
+
+	concurrencyLimit int
+	logger           log.Logger
+	bkt              objstore.Bucket
+}
+
+func NewMetadataFetcher(lg log.Logger, concurrencyLimit int, bkt objstore.Bucket) *MetadataFetcher {
```

Comment:
```suggestion
func NewMetadataFetcher(logger log.Logger, concurrencyLimit int, bkt objstore.Bucket) *MetadataFetcher {
```
Super minor nit, but `logger` is consistent across codebase.